### ISSUE TITLE
No binary cache on hydra.nixos.org

### ISSUE THREAD

**Issue by steshaw** [NixOS/hydra#305]
Hi, I was wondering if there is a problem with hydra.nixos.org at the moment as I have a build that attempts to access http://hydra.nixos.org/nix-cache-info which gives a 404 "There is no binary cache here.".

This happens when building my hakyll-based blog. If it's helpful [the failing log is here](https://circleci.com/gh/steshaw/steshaw.org/477). The build fails because it cannot download hakyll-4.7.

Note that http://hydra.nixos.org told me to log the issue here.

Cheers,
Steve.

**Comment by joachifm** [comment]
My understanding is that build artifacts are copied to https://cache.nixos.org immediately, so there's no point in using hydra.nixos.org any more.

**Comment by steshaw** [comment]
I removed hydra.nixos.org from my list of binary caches. Turns out that this was not really the problem. Somehow the default hakyll had upgraded from 4.7.x to 4.8.x which my code doesn't build against. Unfortunately cabal2nix doesn't remember the versions of the packages I specify in my .cabal file. I ended up generating a shell.nix which I [manually edited](https://github.com/steshaw/steshaw.org/commit/a80fef045099b387e5bab95e73c672a9fa54bc46) to specify lts-5_15 package set (rather than ghc7103). Something like this:

```nix
  haskellPackages = if compiler == "default"
                       then pkgs.haskell.packages.lts-5_15
                       else pkgs.haskell.packages.${compiler};
```
